save/src/lib.rs

```diff
@@ -174,7 +183,24 @@ impl Writer<'_, '_> {
             .map_err(zstd::get_error_name)
             .unwrap();
 
-        self.table.insert(node_id, &*self.compressed)?;
+        self.nodes.insert(node_id, &*self.compressed)?;
+        Ok(())
+    }
+
+    pub fn set_characters(&mut self, characters: Vec<Character>) -> Result<(), redb::Error> {
+        self.meta_value.characters = characters;
+
+        self.plain.clear();
+        self.meta_value.encode(&mut self.plain).unwrap();
+        self.compressed.clear();
+        self.compressed
+            .reserve(zstd::compress_bound(self.plain.len()));
+        self.cctx
+            .compress2(&mut self.compressed, &self.plain)
+            .map_err(zstd::get_error_name)
+            .unwrap();
+
+        self.meta.insert(&[][..], &*self.compressed)?;
```

Comment:
This code looks the same as the code for `put` along with some of the code in `Save::open`. Should we extract this into a function?

Reply:
Done.